auto_round/quantizer.py

```diff
@@ -44,56 +45,270 @@ def reshape_tensor(v, group_size=-1):
 
-    Returns:
-        Quantized and dequantized weight, scale, zero-point
+    Args:
+        orig_layer (torch.nn.Module): The original layer to be wrapped (linear or conv1d).
+        enable_minmax_tuning (bool): Whether to enable min-max scale tuning.
+        enable_norm_bias_tuning (bool): Whether to enable normalization and tuning of the bias term.
+        device (str): Device on which to run computations (e.g., 'cpu' or 'cuda').
     """
-    orig_shape = data.shape
-    if len(data.shape) > 2:
-        data = data.reshape(-1, orig_shape[-1])
-    if group_size == -1 or data.shape[1] < group_size:
-        data, scale, zp = quant_func(data, bits, v=v, min_scale=min_scale, max_scale=max_scale,
-                                     scale_dtype=scale_dtype, weight_min=weight_min, weight_max=weight_max,
-                                     q_scale_thresh=q_scale_thresh, **kwargs)
-        data = data.reshape(orig_shape)
-        return data, scale, zp
-
-    if data.shape[1] % group_size == 0:
-        data = data.reshape(-1, group_size)
-        data, scale, zp = quant_func(data, bits, v=v, min_scale=min_scale, max_scale=max_scale,
-                                     scale_dtype=scale_dtype, weight_min=weight_min, weight_max=weight_max,
-                                     q_scale_thresh=q_scale_thresh, **kwargs)
-        data = data.reshape(orig_shape)
-        return data, scale, zp
 
-    else:
-        tmp_shape = data.shape
-        pad_len = (data.shape[1] + group_size - 1) // group_size * group_size - data.shape[1]
-        data_new = torch.nn.functional.pad(data, (0, pad_len))
-        data_new = data_new.reshape(-1, group_size)
-        data_new, scale, zp = quant_func(data_new, bits, v=v, min_scale=min_scale,
-                                         max_scale=max_scale, scale_dtype=scale_dtype, weight_min=weight_min,
-                                         weight_max=weight_max, q_scale_thresh=q_scale_thresh, **kwargs)
-        data_new = data_new.reshape(tmp_shape[0], -1)
-        data_new = data_new[:, :-pad_len]
-        data_new = data_new.reshape(orig_shape)
-        return data_new, scale, zp
+    def __init__(self, orig_layer, enable_minmax_tuning=True, enable_norm_bias_tuning=False, device='cpu'):
+        """Initializes the WrapperLinear module.
+
+        Args:
+            orig_layer (torch.nn.Module): The original layer to wrap.
+            enable_minmax_tuning (bool): Whether to enable min-max scale tuning.
+            enable_norm_bias_tuning (bool): Whether to enable normalization and tuning for the bias term.
+            device (str): The computation device, such as 'cpu' or 'cuda'.
+        """
+        super(WrapperLinear, self).__init__()
+        self.orig_layer = orig_layer
+        self.device = device
+        self.enable_minmax_tuning = enable_minmax_tuning
+        self.enable_norm_bias_tuning = enable_norm_bias_tuning and orig_layer.bias is not None
```

Comment:
enable_norm_bias_tuning and (orig_layer.bias is not None)

Reply:
not a must, right?  `is not` should be calculated first